cds_migrator_kit/rdm/records/transform/xml_processing/rules/thesis.py

```diff
@@ -50,3 +51,83 @@ def collection(self, key, value):
     if value.get("a").strip().lower() != "thesis":
         raise UnexpectedValue(subfield="a", key=key, value=value, field="690C_")
     raise IgnoreKey("collection")
+
+
+@model.over("publication_date", "(^260__)|(^250__)")
+def imprint_info(self, key, value):
+    """Translates imprint - WARNING - also publisher and publication_date.
+
+    In case of summer student notes this field contains only date
+    but it needs to be reimplemented for the base set of rules -
+    it will contain also imprint place
+    """
+    publication_date_str = value.get("c")
+    _publisher = value.get("b")
+    place = value.get("a")
+    _custom_fields = self.get("custom_fields", {})
+    imprint = _custom_fields.get("imprint:imprint", {})
+
+    if _publisher and not self.get("publisher"):
+        self["publisher"] = _publisher
+
+    if key.startswith("250"):
+        edition = StringValue(value.get("a")).parse()
+        imprint["edition"] = edition
+    if place:
+        imprint["place"] = place
+
+    self["custom_fields"]["imprint:imprint"] = imprint
+    try:
+        date_obj = parse(publication_date_str)
+        return date_obj.strftime("%Y-%m-%d")
+    except (ParserError, TypeError) as e:
+        raise UnexpectedValue(
+            field=key,
+            value=value,
+            message=f"Can't parse provided publication date. Value: {publication_date_str}",
+        )
+
+
+@model.over("custom_fields", "(^502__)")
+def thesis(self, key, value):
+    """Translates custom thesis fields."""
+    _custom_fields = self.get("custom_fields", {})
+    thesis_fields = _custom_fields.get("thesis:thesis", {})
+    dates = self.get("dates", [])
+
+    th_type = StringValue(value.get("a")).parse()
+    uni = StringValue(value.get("b")).parse()
+
+    thesis_fields["type"] = th_type
+    thesis_fields["university"] = uni
+
+    _custom_fields["thesis:thesis"] = thesis_fields
+
+    submission_date = value.get("c")
+    try:
+        submission_date = parse(submission_date)
+        dates.append({"date": submission_date, "type": "submitted"})
+    except (ParserError, TypeError) as e:
+        raise UnexpectedValue(
+            field=key,
+            value=value,
+            message=f"Can't parse provided publication date. Value: {submission_date}",
+        )
+
+    return _custom_fields
+
+
+@model.over("custom_fields", "(^773__)")
+def journal(self, key, value):
+    _custom_fields = self.get("custom_fields", {})
+    journal_fields = _custom_fields.get("journal:journal", {})
+
+    # year = StringValue(value.get("y")).parse()
```

Comment:
leftover?

Reply:
no, this is to be decided by the library, keeping it here for later